modules/http/http-curl-compression.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 One Identity LLC.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef SYSLOG_NG_HTTP_CURL_COMPRESSION_H
+#define SYSLOG_NG_HTTP_CURL_COMPRESSION_H
+
+/* Uncomment upon implementation */
+#define CURL_COMPRESSION_IMPLEMENTED_GZIP
+#define CURL_COMPRESSION_IMPLEMENTED_DEFLATE
+//#define CURL_COMPRESSION_IMPLEMENTED_BROTLI
+//#define CURL_COMPRESSION_IMPLEMENTED_ZSTD
```

Comment:
This looks really weird to me.

Why do we make the code more complicated with these flags and also introduce dead code (in a later commit), when we could just implement what we can now and add the rest later? I we wanted to add something that reminds us that there are functionalities left to implement, why no just put 2 lines of TODO comments before `CurlCompressionTypes` and never even introduce their enums, string representations, empty compression functions, etc?

Reply:
Done